config/application.rb

```diff
@@ -25,6 +25,7 @@
 require_relative '../lib/exceptions'
 require_relative '../lib/sanitize_ext/sanitize_config'
 require_relative '../lib/redis/namespace_extensions'
+require_relative '../lib/nokogiri_handler'
```

Comment:
I probably don't understand the criteria for using either `lib/` or `app/lib/`. The Nokogiri handler is generic, non-Rails-specific code, and is in that respect similar to e.g. the Redis namespace extension, so I assumed it should go into `lib/`.

What is the criteria exactly?

Reply:
The `lib/` and `app/lib/` split is mostly about autoloading / eager-loading of classes. Code in `lib/` is not autoloaded, code in `app/lib/` is. Autoloading not only means you can omit `require` but also that files get reloaded automatically in `development` when they have changed.

This works well for stand-alone classes like this one, but can lead to problems with code that patches existing classes, especially if those are out of the autoloader's scope. That is why you will find mostly patches for gems we use in `lib/`.

I vaguely remember the README of zeitwerk (the aforementioned autoloader, https://github.com/fxn/zeitwerk) explaining all this very well, but at first glance I cannot seem to find the relevant section. It might have been moved to something rails-specific, or I might misremember and it was in an issue or something :confused:

Reply:
Got it :-) I have moved the file to `app/lib/`.